lib/web/fetch/index.js

```diff
@@ -1082,39 +1082,17 @@ function fetchFinale (fetchParams, response) {
   } else {
     // mcollina: all the following steps of the specs are skipped.
     // The internal transform stream is not needed.
-    //
+    // See https://github.com/nodejs/undici/pull/3093#issuecomment-2050198541
+
     // 1. Let transformStream be a new TransformStream.
     // 2. Let identityTransformAlgorithm be an algorithm which, given chunk, enqueues chunk in transformStream.
     // 3. Set up transformStream with transformAlgorithm set to identityTransformAlgorithm and flushAlgorithm
     //    set to processResponseEndOfBody.
     // 4. Set internalResponse’s body’s stream to the result of internalResponse’s body’s stream piped through transformStream.
 
-    const byteStream = new ReadableStream({
-      readableStream: internalResponse.body.stream,
-      async start () {
-        this._bodyReader = this.readableStream.getReader()
-      },
-      async pull (controller) {
-        while (controller.desiredSize >= 0) {
-          const { done, value } = await this._bodyReader.read()
-
-          if (done) {
-            queueMicrotask(() => {
-              readableStreamClose(controller)
-
-              // Added to in lieu of steps 1-4 of the spec
-              processResponseEndOfBody()
-            })
-            break
-          }
-
-          controller.enqueue(value)
-        }
-      },
-      type: 'bytes'
+    finished(internalResponse.body.stream, () => {
```

Comment:
why not 

```js
finished(internalResponse.body.stream, processResponseEndOfBody)
```

why the addional function around it?

Reply:
For safety, this approach won't break if processResponseEndOfBody ever takes an argument. No reason to micro-optimize this.

Reply:
makes sense. i would optimize it, but it makes sense.